x-pack/plugins/security_solution/public/plugin_services.ts

```diff
@@ -129,7 +129,7 @@ export class PluginServices {
       telemetry: this.telemetry.start(),
       customDataService,
       topValuesPopover: new TopValuesPopoverService(),
-      timelineFilterManager: new FilterManager(coreStart.uiSettings),
+      timelineDataService: this.startTimelineDataService(query, startPlugins.data),
```

Comment:
I see one problem with this. `timelineDataservice` and `customDataService` (line 113) share the `query` instance which also means it shares the filter manager.

`customDataService` is used by discover i.e ES|QL tab, which when modifies the filter, will impact the filters of timeline.

I think query instance should be separate between both data service.

Reply:
Okay I just did the desk testing for this particular scenario, I am not able to reproduce this because we use `esql` only and may it does not use the same service and it does not even use filter as well... so I am not sure why it is not conflicting but it should 🤣

Reply:
Good catch @logeekal, I don't know exactly what the filterManager is used for inside the ES|QL logic, I don't see any way to add filters (may be just my ignorance), but what I see is the "Query" filterManager being reset when the "ES|QL" tab renders:

https://github.com/elastic/kibana/assets/17747913/5f45ca71-d5a0-4a07-9fc1-fa41ccd6674b

We should consider using a different filterManager on each tab.

Reply:
I tried to fix it with this: https://github.com/elastic/kibana/pull/180850/commits/e9a214717429820f0f05c7d1fddfabb35f884f12 but that didn't work. WIll investigate more.

Reply:
@janmonschke I tried creating a separate QueryService for timeline at line 34:
```
  private readonly queryService: QueryService = new QueryService();
  private readonly timelineQueryService: QueryService = new QueryService();
```
Calling `setup` and `start` the same way (`start` should probably moved inside `startCustomDataService` and `startTimelineDataService` accordingly)

And it worked.

Reply:
https://github.com/elastic/kibana/pull/180850/commits/656b336baf46770ba98d771f3ba945a42ffd6901 👍

Reply:
Given this issue exists in `main` (as described in https://github.com/elastic/kibana/issues/181666), it should unblock this PR. Wdyt @logeekal ?

Reply:
I agree. This PR looks good to me. Thanks for making the changes @janmonschke